csharp/src/ProtocolBuffers/FieldAccess/FieldAccessorTable.cs

```diff
@@ -84,7 +95,7 @@ public FieldAccessorTable(Type type, MessageDescriptor descriptor, string[] prop
             }
         }
 
-        internal IFieldAccessor this[FieldDescriptor field]
+        public IFieldAccessor this[FieldDescriptor field]

```

Comment:
this is used just once, and it can be replaced by [field.Index] in that case. I suggest removing this for now until we figure out how to address the TODO above.